Swiften/Parser/PayloadParsers/ClientStateParser.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta.
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/Parser/PayloadParsers/ClientStateParser.h>
+
+namespace Swift {
+
+ClientStateParser::ClientStateParser() : level_(0) {
+}
+
+void ClientStateParser::handleStartElement(const std::string& element, const std::string&, const AttributeMap&) {
+    if (level_ == 0) {
+        ClientState::ClientStateType state = ClientState::Active;
```

Comment:
You can use **auto** on the left side as a type and have the compiler deduce the type.